step-release-vis/src/app/services/environment.ts

```diff
@@ -62,13 +201,66 @@ export class EnvironmentService {
   }
 ]
 `;
+    json = `[
+  {
+    "environment": "prod",
+    "snapshots": [
+      {
+        "timestamp": 1,
+        "cands_info": [
+          {
+            "name": "1",
+            "job_count": 10
+          },
+          {
+            "name": "2",
+            "job_count": 90
+          }
+        ]
+      },
+            {
+        "timestamp": 2,
+        "cands_info": [
+          {
+            "name": "1",
+            "job_count": 90
+          },
+          {
+            "name": "2",
+            "job_count": 10
+          }
+        ]
+      }
+    ]
+  }
+]`;
     const envs: Environment[] = JSON.parse(json);
-    return [];
+    return envs;
   }
 }
 
 type CandidatesSnapshot = CandidateSnapshot[];
 
+class PointSet {
```

Comment:
both these helper classes are a bit hard to understand, I think I more or less get, what you want to use them for, but it is not trivial based on the code, either restructure, or document them better ;)